File: 백준/Bronze/25304. 영수증/영수증.py
total = int(input())
kind = int(input())

buy_list = []

for i in range(0,kind):
    a,b = map(int,input().split())
    buy_list.append(a*b)

if total == sum(buy_list):
    print("Yes")
else:
    print("No")

File: 백준/Bronze/25304. 영수증/영수증.go
package main

import (
	"fmt"
)

func main() {
	var total, kind int
	var price, amount int

	fmt.Scanln(&total)
	fmt.Scanln(&kind)

	ProdPriceList := make([]int, kind)

	for i := 0; i < kind; i++ {
		fmt.Scanln(&price, &amount)
		ProdPriceList[i] = price * amount
	}

	sum := 0

	for _, ProdPrice := range ProdPriceList {
		sum += ProdPrice
	}

	if sum == total {
		fmt.Print("Yes")
	} else {
		fmt.Print("No")
	}

}
